src/diag/kwp2000.cpp

```diff
@@ -449,26 +449,42 @@ void Kwp2000_server::process_read_data_local_ident(uint8_t* args, uint16_t arg_l
         sprintf(resp, "%02X%02X%02X%02X%02X%02X", mac[0], mac[1], mac[2], mac[3], mac[4], mac[5]);
         make_diag_pos_msg(SID_READ_DATA_LOCAL_IDENT, 0xE1, (const uint8_t*)resp, 12);
     } else if (args[0] == RLI_MAP_EDITOR) {
-        // We need 2 args for this. 
-        if (arg_len != 2) {
-            make_diag_neg_msg(SID_READ_DATA_LOCAL_IDENT, NRC_SUB_FUNC_NOT_SUPPORTED_INVALID_FORMAT);
-            return;
-        }
-        int16_t* read_buf = nullptr;
-        uint16_t dest_size = 0;
-        uint8_t ret = MapEditor::read_map_data(args[1], &dest_size, &read_buf);
-        if (ret == 0) { // OK
-            uint8_t* buf = new uint8_t[2+dest_size];
-            buf[0] = dest_size >> 8;
-            buf[1] = dest_size & 0xFF;
-            memcpy(&buf[2], read_buf, dest_size);
-            make_diag_pos_msg(SID_READ_DATA_LOCAL_IDENT, buf, 2+dest_size);
-            delete read_buf; // DELETE MapEditor allocation
-            return;
-        } else {
-            make_diag_neg_msg(SID_READ_DATA_LOCAL_IDENT, ret);
-            return;
-        }
+            // 0 - RLI
+            // 1 - Map ID
+            // 2 - CMD
+            // 3-4 - Arg len
+            // 5..n - Data
+            if (arg_len < 5) {
```

Comment:
Values, which cannot be changed, should generally be on the left hand side in order to avoid an assignment by accident